### ISSUE TITLE
What is the proper configs?

### ISSUE THREAD

**Issue by seanappler** [THUcqb/KOBE#5]
Thanks for sharing the code.
In the config file, learning_rate is set to 2, but in the paper, saying that learning_rate is set to 10^-4.
I tried both settings for baseline, and founded 10^-4 was too small to improve BLEU score.  But for learning_rate=2, BLEU score was about 6.0 last 1M steps which is not 7.2 showed in the paper. Did I make any mistakes, thx.

**Comment by qibinc** [comment]
Hi @seanappler ,

Thanks for your interest in this work!

#### learning rate

For learning_rate, we set the initial learning rate to 2 and have a decay method described here:

https://github.com/THUDM/KOBE/blob/48b321d6fb771f71938a002da404569be0939864/core/models/optims.py#L51-L66

We found this saves time for training so we added it in this version. To fully reproduce the paper version (1e-4) results, please comment out these lines and set the learning_rate to 1e-4 in the config files.

#### BLEU score

In the paper, the results are computed based on beam-searched results. However, in training, the logged BLEU is computed by greedy decoding. This causes the difference you pointed out.

To run beam search with a trained model, please pull the latest code and run this example:

```bash
python core/train.py --config configs/baseline.yaml \
--mode eval --restore experiments/baseline/checkpoint.pt \
--expname eval-baseline \
--beam-size 10
```

Hope this helps.

Best,
Qibin Chen

**Comment by seanappler** [comment]
Thx a lot!!! I figured it out.
